services/ingest/tests/test_correlation_engine.py

```
@@ -370,3 +430,62 @@ async def test_skips_image_tag_when_not_provided(self):
+class TestApplyTerminalGuardedStatus:
+    """/code-review high + CodeRabbit on PR #796: this must be a
+    WHERE-guarded UPDATE, not an in-memory ORM mutation, or a stale
+    delivery can overwrite a status a concurrent transaction already
+    committed to terminal between the caller's correlating SELECT and
+    this call."""
+
+    @pytest.mark.asyncio
+    async def test_applies_when_not_terminal(self):
+        captured = []
+
+        async def mock_execute(stmt):
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `mock_execute`

(ANN202)